### DISCUSSION TITLE
Any templates for same workflow, multiple virtual environments

### DISCUSSION THREAD

**Discussion by ajhalili2006** [community/community#27060] [Actions]
<p>Hi GitHub Community,</p>
<p>Is there any available workflow templates if I want to test my scripts on both Ubuntu and macOS Catalina?</p>

**Comment by Simran-B** [comment, marked as answer]
<p>Hi, did you find what you were looking for in the meantime? It’s not fully clear to me from your question what you look for, but it sounds like you want a single workflow to run with different runners (Ubuntu, macOS). That’s easily achievable with a <a href="https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstrategymatrix" rel="noopener nofollow ugc">job matrix</a>:</p>
<pre><code class="lang-yaml">on: push
jobs:
  job1:
    runs-on: ${{ matrix.os }}
    strategy:
      matrix:
        os: [ubuntu-latest, macos-latest]
    steps:
      - name: Run this for all OSes
        run: |
          echo "os = ${{ matrix.os }}"
      - name: Run this on Linux only
        if: contains(matrix.os, 'ubuntu-')
        run: |
          echo "Linux only"
      - name: Run this on macOS only
        if: contains(matrix.os, 'macos-')
        run: |
          echo "macOS only"
</code></pre>
<p>You need to make sure that the code you run works on all runners. You may want to set <code>shell: bash</code> for steps so that it’s the same shell across OSes.</p>
